commit e4491cfd74ac1d9e30ab264a66ea27e5837b1919
Author: John Johansen <john.johansen@canonical.com>
Date:   Mon May 23 12:04:57 2016 -0700

    UBUNTU: SAUCE: apparmor: Fix label build for onexec stacking.
    
    The label build for onexec when crossing a namespace boundry is not
    quite correct. The label needs to be built per profile and not based
    on the whole label because the onexec transition only applies to
    profiles within the ns. Where merging against the label could include
    profile that are transitioned via the profile_transition callback
    and should not be in the final label.
    
    BugLink: http://bugs.launchpad.net/bugs/1615881
    
    Signed-off-by: John Johansen <john.johansen@canonical.com>
    Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>

diff --git a/security/apparmor/domain.c b/security/apparmor/domain.c
index 814f8cd..35ed7c2 100644
--- a/security/apparmor/domain.c
+++ b/security/apparmor/domain.c
@@ -644,7 +644,8 @@ static struct aa_label *handle_onexec(struct aa_label *label,
 		if (error)
 			return ERR_PTR(error);
 		new = fn_label_build_in_ns(label, profile, GFP_ATOMIC,
-					   aa_label_merge(label, onexec,
+					   aa_label_merge(&profile->label,
+							  onexec,
 							  GFP_ATOMIC),
 					   profile_transition(profile, xname,
 							      cond, unsafe));
